docs/toggle_course_features_help.md

```diff
@@ -0,0 +1,33 @@
+# Guide to adding a course feature (that can be toggled on/off by the professor)
+
+Current features that can be toggled for a course:
+- Chatbot
+- Async Question Centre
+- Queues
+- Ads
+
+### Backend
+
+- `/common/index.ts`: add the new feature in the validFeatures array and the CourseSettingsResponse type.
+- `course_settings.entity.ts`: Add the new feature as a column and its default value (true/false)
+- `course.controller.ts`: Add the new feature in the @Get(':id/features) endpoint in the two response bodies beside the other features.
+    - sidenote: the entire endpoint exists in the controller, there is nothing for coursefeatures in the `course.service.ts` file, so no changes are necessary there.  
+- `course.integration.ts`: tests for the toggle features endpoint. 
+    - The PATCH test (specifically the "should return 200 if course settings are updated successfully" test) should be updated to include the new feature.
+    - The GET test should be updated to include the new feature (specifically, where it expects the payloads).
+- `organization.integration.ts`: specifically the following tests will need to be updated with the new feature:
+    - "should return 202 when a course is created with no course settings provided (which will use defaults)"
+    - (optionally) "should return 200 when course is created" 
+
+### Frontend
+- `/api-client/index.ts`: no changes needed here. Can use the setCourseFeature and getCourseFeatures functions.
```

Comment:
This is not necessarily frontend but a shared component?

Reply:
yeah idk, I guess you are correct. I guess my thought was that the functions there in index.ts are used by the frontend to call backend APIs but i guess technically the backend can use those functions too?